pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1208,6 +1208,24 @@ func (c *Controller) CreatePodTemplate(
 func (c *Controller) setStorageClusterDefaults(cluster *corev1.StorageCluster) error {
 	toUpdate := cluster.DeepCopy()
 
+	if cluster.Spec.CSI == nil {
```

Comment:
this should be changed in this function instead https://github.com/libopenstorage/operator/blob/312fa642d31e8c0fac62897d3849eef9fe0d89b6/drivers/storage/portworx/portworx.go#L1066
and unit test `SetDefaultsOnStorageCluster` in `component_test.go`